NWTOPT.py

```diff
@@ -20,110 +20,118 @@
 # pylint: disable = E0401, C0103, W0621, R0914, R0912, R0915, R0915
 
 
-import subprocess
 import time
+import sys
 import os
 import socket
 import argparse
-import fileinput
-import signal
-
-## Modify nwtopt.sub to have user inputed ip:port and poll interval
-def modifySubmitFile(workers, ip, port, pollInterval):
-    for line in fileinput.input('nwtopt.sub', inplace = True):
-        if line.startswith('arguments'):
-            print(f'arguments               = {ip}:{port}/db {pollInterval}', end=os.linesep)
-        elif line.startswith('queue'):
-            print(f'queue {workers}', end=os.linesep)
+import logging
+import asyncio
+from collections import defaultdict
+from objects.OPTSubprocess import OPTSubprocess
+from objects.Master import Master
+from objects.DB import DB
+from objects.Condor import Condor
+from objects.DB_Poller import DB_Poller
+
+class NWTOPT():
+    def __init__(self, args):
+        self.ip = args.ip
+        self.port = args.port
+        self.key = args.key
+        self.workers = args.workers
+        self.random = args.random
+        self.trials = args.trials
+        self.poll_interval = args.poll_interval
+        self.enable_condor = args.enable_condor
+        self.timeout = args.timeout
+        self.cwd = os.getcwd()
+        self.processes = defaultdict(OPTSubprocess)
+        self.logger = self.init_logger()
+        
+        self.event_loop = asyncio.new_event_loop()
+        
+        self.log(f'Working out of {self.cwd}', 0)
+
+    def init_logger(self):
+        logger = logging.getLogger('NWTOPT')
+        logger.setLevel(logging.DEBUG)
```

Comment:
I will look as this progresses, but first look suggests maybe this `logger` object could get assigned to the `NWTOPT` object through `self`?